src/ServiceInsight/Explorer/EndpointExplorer/EndpointExplorerViewModel.cs

```diff
@@ -3,203 +3,97 @@
+                root.IsExpanded = true;
 
-            eventAggregator.PublishOnUIThread(new WorkFinished());
-        }
+                Items.Add(root);
+            }
 
-        string GetConfiguredAddress()
-        {
-            if (commandLineParser.ParsedOptions.EndpointUri == null)
+            var toRemove = root.Children.ToList();
+            var endpointInstancesGroupedByName = e.Data.OrderBy(x => x.name).GroupBy(x => x.name);
+            foreach (var scaledOutEndpoint in endpointInstancesGroupedByName)
             {
-                var appSettings = settingsProvider.GetSettings<ProfilerSettings>();
-                if (appSettings != null && appSettings.LastUsedServiceControl != null)
+                var instances = scaledOutEndpoint.ToList();
+                var endpoint = instances.Cast<JObject>().First().ToObject<Models.Endpoint>(serializer);
+                var node = root.GetEndpointNode(endpoint);
+                if (node != null)
                 {
-                    return appSettings.LastUsedServiceControl;
+                    toRemove.Remove(node);
+                    continue;
                 }
 
-                var managementConfig = settingsProvider.GetSettings<ServiceControlSettings>();
-                return string.Format("http://localhost:{0}/api", managementConfig.Port);
-            }
+                var hostNames = instances.Select(instance => instance.host_display_name).Distinct();
+                var tooltip = string.Join(", ", hostNames);
 
-            return commandLineParser.ParsedOptions.EndpointUri.ToString();
+                root.Children.Add(new AuditEndpointExplorerItem(endpoint, tooltip));
+            }
+            if (toRemove.Any())
+            {
+                var saveSelectedNode = SelectedNode;
+                root.Children.RemoveRange(toRemove);
+                SelectedNode = saveSelectedNode;
+            }
         }
 
-        bool ServiceAvailable(string serviceUrl)
-        {
-            connectionProvider.ConnectTo(serviceUrl);
-
-            var connected = serviceControl.IsAlive();
-
-            return connected;
-        }
+        public IObservableCollection<ExplorerItem> Items { get; }
 
-        void AddServiceNode()
-        {
-            Items.Clear();
-            Items.Add(new ServiceControlExplorerItem(ServiceUrl));
-        }
+        public ExplorerItem SelectedNode { get; set; }
 
         public void OnSelectedNodeChanged()
         {
             eventAggregator.PublishOnUIThread(new SelectedExplorerItemChanged(SelectedNode));
         }
 
-        void SelectDefaultEndpoint()
+        public void Handle(RequestSelectingEndpoint message)
         {
-            if (ServiceControlRoot == null)
+            foreach (var item in Items.OfType<ServiceControlExplorerItem>())
             {
-                return;
-            }
-
-            if (!commandLineParser.ParsedOptions.EndpointName.IsEmpty())
```

Comment:
This code was required for when SI is activated via a link, e.g. si://ServiceControlUrl/endpoint and it'd focus on that particular endpoint. Was it removed intentionally?